lib/shelley/bench/restore-bench.hs

```diff
@@ -596,6 +623,53 @@ benchmarksSeq _ w wid _wname benchname restoreTime = do
         , walletOverview
         }
 
+data BenchBaselineResults = BenchBaselineResults
+    { benchName :: Text
+    , restoreTime :: Time
+    } deriving (Show, Generic)
+
+instance Buildable BenchBaselineResults where
+    build = genericF
+
+instance ToJSON BenchBaselineResults where
+    toJSON = genericToJSON Aeson.defaultOptions
+
+{- HLINT ignore bench_baseline_restoration "Use camelCase" -}
+bench_baseline_restoration
+    :: forall (n :: NetworkDiscriminant) .
+        (
+         NetworkDiscriminantVal n
+        , HasNetworkId n
+        )
+    => Proxy n
+    -> Tracer IO (BenchmarkLog n)
+    -> Trace IO Text -- ^ For wallet tracing
+    -> CardanoNodeConn  -- ^ Socket path
+    -> NetworkParameters
+    -> NodeToClientVersionData
+    -> Text -- ^ Benchmark name (used for naming resulting files)
+    -> Bool -- ^ if to write to disk
+    -> Percentage -- ^ Target sync progress
+    -> IO SomeBenchmarkResults
+bench_baseline_restoration proxy tr wlTr socket np vData benchname traceToDisk  targetSync = do
```

Comment:
This function is very wide. I would recommend placing the parameters on a separate line, or possibly introducing a record type to capture all the parameters!

```suggestion
bench_baseline_restoration
    proxy tr wlTr socket np vData benchname traceToDisk targetSync = do
```